pkgs/development/compilers/llvm/12/compiler-rt/default.nix

```diff
@@ -91,6 +93,8 @@ stdenv.mkDerivation {
     ln -s $out/lib/*/clang_rt.crtend-*.o $out/lib/crtend.o
     ln -s $out/lib/*/clang_rt.crtbegin_shared-*.o $out/lib/crtbeginS.o
     ln -s $out/lib/*/clang_rt.crtend_shared-*.o $out/lib/crtendS.o
+  '' + lib.optionalString (doFakeLibgcc) ''
+    ln -s $out/lib/freebsd/libclang_rt.builtins-*.a $out/lib/libgcc.a
```

Comment:
Hm, does that mean `doFakeLibgcc` only works on FreeBSD?

Reply:
It is just used for FreeBSD right now. I wish we didn't have to do this but the FreeBSD Clang frontend just has these sorts of things hard-coded because that's how FreeBSD currently bootstrap things.

The larger issue is the "OS" field of the target name really has nebulous meaning (is it the syscall ABI, the libc ABI, something more? etc.) and so we get these postels law gone wrong situations of people hacking in ad-hoc special cases to get things working that just cause extra headache for something like Nixpkgs where we rather do all the combinations as uniformly as possible.